File: benchmarks/cpp/map.cpp
#include <cstdio>
#include <string>
#include <unordered_map>
int main() {
    std::unordered_map<std::string, long long> counts;
    for (long long i = 0; i < 300000; i++) counts["key" + std::to_string(i)] = i;
    long long sum = 0;
    for (long long i = 0; i < 300000; i++) sum += counts["key" + std::to_string(i)];
    printf("%zu %lld\n", counts.size(), sum);
}

File: benchmarks/rust/map.rs
use std::collections::HashMap;
fn main() {
    let mut counts: HashMap<String, i64> = HashMap::new();
    for i in 0..300_000i64 { counts.insert(format!("key{}", i), i); }
    let mut sum: i64 = 0;
    for i in 0..300_000i64 { sum += counts[&format!("key{}", i)]; }
    println!("{} {}", counts.len(), sum);
}
